src/convertToObject.js

```diff
@@ -3,16 +3,31 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const allEntries = sourceString.split(';');
+  const objectEntries = allEntries.filter(entry => entry.trim());
+  const object = objectEntries
+    .reduce(
+      (prev, entry, i) => {
+        const key = entry.slice(0, entry.indexOf(':'));
+        const value = entry.slice(entry.indexOf(':') + 1);
+
+        return {
+          ...prev, [key.trim()]: value.trim(),
+        };
+      }, {});
+
+  return object;
```

Comment:
good job.
just want to show little another solution.
filtering looks like redundant and possible make all in reduce.
```suggestion
  return allEntries
    .reduce(
      (prev, entry, i) => {
        const [key, value] = entry.split(':').map(item => item.trim());

        if (!key) {
          return prev;
        }

        return {
          ...prev, [key]: value,
        };
      }, {});
```